crates/solver/src/liquidity_collector.rs

```diff
@@ -82,7 +82,13 @@ impl<L> BackgroundInitLiquiditySource<L> {
                             tokio::time::sleep(retry_init_timeout).await;
                         }
                         Ok(source) => {
-                            let _ = inner.lock().await.insert(source);
+                            if inner.set(source).is_err() {
+                                // should never happen but if it does we want to know about it ASAP
+                                tracing::error!(
+                                    source = inner_label,
+                                    "liquidity source already initialized"
+                                );
+                            }
```

Comment:
nit: I understand that should never happen, but wouldn't it be slightly clearer that the metrics are incremented only when the `OnceCell::set` succeeds?
```suggestion
                             if inner.set(source).is_err() {
                                // should never happen but if it does we want to know about it ASAP
                                tracing::error!(
                                    source = inner_label,
                                    "liquidity source already initialized"
                                );
                            } else {
                                tracing::debug!("successfully initialised liquidity source");
                                Metrics::get()
                                    .liquidity_enabled
                                    .with_label_values(&[&inner_label])
                                    .inc();
                            }
```

Reply:
Yeah, that's fair. 👍